plugins/notifications-backend-module-slack/src/lib/SlackNotificationProcessor.ts

```diff
@@ -0,0 +1,251 @@
+    logger: LoggerService;
+    directWebhookUrl?: string;
+    channelWebhookUrl?: string;
+    catalogClient: CatalogClient;
+  }) {
+    const { auth, catalogClient, directWebhookUrl, channelWebhookUrl, logger } =
+      options;
+    this.logger = logger;
+    this.directWebhookUrl = directWebhookUrl;
+    this.channelWebhookUrl = channelWebhookUrl;
+    this.catalogClient = catalogClient;
+    this.auth = auth;
+
+    // https://api.slack.com/apis/rate-limits
+    const throttle = pThrottle({
+      limit: 10,
+      interval: durationToMilliseconds({ minutes: 1 }),
+    });
+    const throttled = throttle(opts => sendSlackNotification(opts));
+    this.sendSlackNotifications = async (opts: SlackNotificationOptions[]) =>
+      await Promise.all(opts.map(message => throttled(message)));
+  }
+
+  getName(): string {
+    return 'Slack';
+  }
+
+  async processOptions(
+    options: NotificationSendOptions,
+  ): Promise<NotificationSendOptions> {
+    if (!this.channelWebhookUrl || options.recipients.type !== 'entity') {
+      return options;
+    }
+
+    const entityRefs = [options.recipients.entityRef].flat();
+
+    const outbound: SlackNotificationOptions[] = [];
+    await Promise.all(
+      entityRefs.map(async entityRef => {
+        const compoundEntityRef = parseEntityRef(entityRef);
+        // skip users as they are sent direct messages, but allow all other entity kinds
+        // to have a channel id annotation.
+        if (compoundEntityRef.kind === 'user') {
+          return;
+        }
+
+        let destination;
+        try {
+          destination = await this.getChannelFromEntity(entityRef);
+        } catch (error) {
+          this.logger.error(
+            `Failed to get Slack channel for entity: ${error.message}`,
+          );
+          return;
+        }
+
+        if (!destination) {
+          this.logger.debug(`No Slack channel found for entity: ${entityRef}`);
+          return;
+        }
+
+        const payload = JSON.stringify({
+          ...options.payload,
+          destination,
+        });
+
+        this.logger.debug(`Sending Slack channel notification: ${payload}`);
+        outbound.push({ url: this.channelWebhookUrl!, payload });
+      }),
+    );
+
+    await this.sendSlackNotifications(outbound).catch(error => {
+      this.logger.error(
+        `Failed to send Slack channel notification: ${error.message}`,
+      );
+    });
+
+    if (outbound.length > 0) {
+      // If we sent team notifications, we don't want to send individual ones after resolving the group
+      // This is how we prevent the other methods from sending individual notifications.
+      // Given the current logic, if a Group has no slack channel id annotation, we'll let it get expanded to individuals.
+      options.recipients.entityRef = [];
+    }
+    return options;
+  }
+
+  async postProcess(
+    notification: Notification,
+    options: NotificationSendOptions,
+  ): Promise<void> {
+    if (
```

Comment:
For better readability this triple condition could might be explained via extra variable, for example (or some other name like this):

```
const canSendDirectMessage = !this.directWebhookUrl ||
      options.recipients.type === 'broadcast' ||
      !notification.user;
```

Otherwise I'm as a code reader can only suggest the intention there and takes time to understand the purpose of filtering out the notification.